proxy/envoy/policy.go

```diff
@@ -105,18 +106,28 @@ func insertDestinationPolicy(config *model.IstioRegistry, cluster *Cluster) {
 			cluster.OutlierDetection = &OutlierDetection{}
 
 			cluster.OutlierDetection.MaxEjectionPercent = 10
-			if cbconfig.SleepWindowSeconds > 0 {
-				cluster.OutlierDetection.BaseEjectionTimeMS = int(cbconfig.SleepWindowSeconds * 1000)
+			if cbconfig.SleepWindow.Seconds > 0 {
+				cluster.OutlierDetection.BaseEjectionTimeMS = protoDurationToMS(cbconfig.SleepWindow)
 			}
 			if cbconfig.HttpConsecutiveErrors > 0 {
 				cluster.OutlierDetection.ConsecutiveErrors = int(cbconfig.HttpConsecutiveErrors)
 			}
-			if cbconfig.HttpDetectionIntervalSeconds > 0 {
-				cluster.OutlierDetection.IntervalMS = int(cbconfig.HttpDetectionIntervalSeconds * 1000)
+			if cbconfig.HttpDetectionInterval.Seconds > 0 {
+				cluster.OutlierDetection.IntervalMS = protoDurationToMS(cbconfig.HttpDetectionInterval)
 			}
 			if cbconfig.HttpMaxEjectionPercent > 0 {
 				cluster.OutlierDetection.MaxEjectionPercent = int(cbconfig.HttpMaxEjectionPercent)
 			}
 		}
 	}
 }
+
+func protoDurationToMS(dur *duration.Duration) int {
```

Comment:
please use convertDuration to avoid manipulating time directly. You can divide by unit to get MS.